### ISSUE TITLE
SocketIO deployment enters crashloopbackoff when using external redis deployment

### ISSUE THREAD

**Issue by bpoulse** [frappe/helm#60]
## Description of the issue
Helm chart socketio deployment crashes when trying to use an existing redis deployment
## Steps to reproduce the issue

1. Install MariaDB and Redis **externally from erpnext**
2. Install ERPnext using the helm chart and the following values for Redis:
```
redisQueueHost: "redis-master.default.svc.cluster.local"
redisCacheHost: "redis-master.default.svc.cluster.local"
redisSocketIOHost: "redis-master.default.svc.cluster.local"
```
### Observed result
SocketIO deployment enters crash loop backoff because it still expects the following host: `erpnext-redis-socketio` not the one specified in the helm chart. This is also the case if the chart is named anything other than "erpnext".
### Expected result
Chart deploys successfully
### Stacktrace / full error message if available

```
listening on *: 9000
node:events:356
      throw er; // Unhandled 'error' event
      ^

Error: getaddrinfo ENOTFOUND erpnext-redis-socketio
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:69:26)
Emitted 'error' event on RedisClient instance at:
    at RedisClient.on_error (/home/frappe/frappe-bench/apps/frappe/node_modules/redis/index.js:406:14)
    at Socket.<anonymous> (/home/frappe/frappe-bench/apps/frappe/node_modules/redis/index.js:279:14)
    at Socket.emit (node:events:379:20)
    at emitErrorNT (node:internal/streams/destroy:188:8)
    at emitErrorCloseNT (node:internal/streams/destroy:153:3)
    at processTicksAndRejections (node:internal/process/task_queues:81:21) {
  errno: -3008,
  code: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'erpnext-redis-socketio'
}
```

**Comment by revant** [comment]
I am able to use separate Elasticache in one of the clusters.
On another cluster I've setup separate redis server manually and I can access it by private ip.

Check networking of the external helm chart, does it need extra config to make it available for access?

**Comment by bpoulse** [comment]
I believe I have solved it. I did not realize that the env variables are saved on the first deployment and not used afterward (documented in the docker images). This would be helpful to document in this repo. I was using an AWS EFS storageclass (to achieve ReadWriteMany on sites_dir) that persisted after a `helm uninstall`. Originally I installed the helm chart as default (with the redis deployments) and only afterward switched to an external Redis deployment.
